packages/wallet/test/e2e/config.ts

```diff
@@ -145,7 +145,14 @@ export const keyAgentByIdx = async (accountIndex: number) => {
   }
 };
 
-export const keyAgentReady = (() => keyAgentByIdx(0))();
+export const keyAgentReady = (() => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const keyAgent = keyAgentByIdx(0) as any;
+  if (process.env.KEY_AGENT === 'Ledger' && keyAgent.deviceConnection) {
+    deviceConnection = keyAgent.deviceConnection;
```

Comment:
Isn't this redundant as `keyAgentByIdx` sets it?

Reply:
Note: Potential solution will be to memoize keyAgentByIdx(0) like we discussed.

Reply:
✅